init.lua

```diff
@@ -844,7 +844,12 @@ textadept.editing.autocompleters.lsp = function()
 			label = symbol.filterText or symbol.label
 			snippets[label] = symbol.insertText
 		end
-		-- TODO: some labels can have spaces and need proper handling.
+
+		if string.find(label, ' ') then
+			-- A random value to ensure it never gets used.
+			buffer.auto_c_separator = string.byte('81bdf528d67a87231dff9da2f746fe')
```

Comment:
Unfortunately this is not an appropriate solution. `buffer.auto_c_separator` will be the character '8', so a list with the contents ["i", "j81", "k2"] will actually be presented as a list of ["ij", "1k2"].

This is because `textadept.editing.autocomplete()` takes a list, concatenates it into a single string separated by `buffer.auto_c_separator` characters, and then passes that string to `buffer:auto_c_show()`, which in turn splits that parameter by `buffer.auto_c_separator` characters to produce the list. It's quite convoluted, but that's how Scintilla's API works for this method. (Scintilla is the editor component Textadept uses.)

You can probably get away with setting `buffer.auto_c_separator = string.byte('|')`, but I haven't fully explored how best to handle this because it would end up affecting all autocomplete usage in one way or another. (Hence the TODO...)

Reply:
In a solution I devised before this, it replaced all spaces with the non-breaking space character. This worked visually. However when the label is changed like this, when the user chooses an item with visual spaces from the autocompletion list, textadept sends the chosen label (which now contains non-breaking spaces) to the lsp and of course the language server has no clue about what that item is.

Maybe a future solution may involve decoupling the presentation (what the user sees) from the items coming out of and being sent to language server. For now I'll use the '|' workaround.